core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -242,8 +243,13 @@ private void writeSortedFile(boolean isLastFile) {
       // Note that we intentionally ignore the value of `writeMetricsToUse.shuffleWriteTime()`.
       // Consistent with ExternalSorter, we do not count this IO towards shuffle write time.
       // This means that this IO time is not accounted for anywhere; SPARK-3577 will fix this.
-      writeMetrics.incRecordsWritten(writeMetricsToUse.recordsWritten());
-      taskContext.taskMetrics().incDiskBytesSpilled(writeMetricsToUse.bytesWritten());
+
+      // This is guaranteed to be a ShuffleWriteMetrics based on the if check in the beginning
+      // of this file.
```

Comment:
I found "beginning of this file" to be confusing, I thought you meant the beginning of `ShuffleExternalSorter.java`, not the spill file. maybe "beginning of this method"

also is the comment above this about SPARK-3577 out of date now that has been fixed?